src/charm.py

```diff
@@ -619,7 +624,7 @@ def _restart(self, _) -> None:
                 for attempt in Retrying(stop=stop_after_attempt(24), wait=wait_fixed(5)):
                     with attempt:
                         if self._mysql.is_instance_in_cluster(unit_label):
-                            self.unit.status = ActiveStatus(self.active_status_message)
+                            self._on_update_status(None)
```

Comment:
Because it's a event handler that's also called within the code, so when called from the event (`update_status`), ops does put a parameter (the event) there.